src/apify_client/clients/resource_clients/dataset.py

```diff
@@ -0,0 +1,377 @@
+                To reverse the order, set this parameter to True.
+            clean (bool, optional): If True, returns only non-empty items and skips hidden fields (i.e. fields starting with the # character).
+                The clean parameter is just a shortcut for skip_hidden=True and skip_empty=True parameters.
+                Note that since some objects might be skipped from the output, that the result might contain less items than the limit value.
+            fields (list[str], optional): A list of fields which should be picked from the items,
+                only these fields will remain in the resulting record objects.
+                Note that the fields in the outputted items are sorted the same way as they are specified in the fields parameter.
+                You can use this feature to effectively fix the output format.
+            omit: (list[str], optional): A list of fields which should be omitted from the items.
+            unwind: (str, optional): Name of a field which should be unwound.
+                If the field is an array then every element of the array will become a separate record and merged with parent object.
+                If the unwound field is an object then it is merged with the parent object.
+                If the unwound field is missing or its value is neither an array nor an object and therefore cannot be merged with a parent object,
+                then the item gets preserved as it is. Note that the unwound items ignore the desc parameter.
+            skip_empty (bool, optional): If True, then empty items are skipped from the output.
+                Note that if used, the results might contain less items than the limit value.
+            skip_hidden (bool, optional): If True, then hidden fields are skipped from the output, i.e. fields starting with the # character.
+
+        Returns:
+            The dataset items
+        """
+        request_params = self._params(
+            offset=offset,
+            limit=limit,
+            desc=desc,
+            clean=clean,
+            fields=fields,
+            omit=omit,
+            unwind=unwind,
+            skipEmpty=skip_empty,
+            skipHidden=skip_hidden,
+        )
+
+        response = self.http_client.call(
+            url=self._url('items'),
+            method='GET',
+            params=request_params,
+        )
+
+        data = response.json()
+
+        return {
+            'items': data,
+            'total': int(response.headers['x-apify-pagination-total']),
+            'offset': int(response.headers['x-apify-pagination-offset']),
+            'count': len(data),  # because x-apify-pagination-count returns invalid values when hidden/empty items are skipped
+            'limit': int(response.headers['x-apify-pagination-limit']),  # API returns 999999999999 when no limit is used
+        }
+
+    def iterate_items(
+        self,
+        offset: int = 0,
+        limit: int = 99999999999,
```

Comment:
I don't like this, but I guess it is an API issue, not the client. Also, there is one 9 less than in the comment a few lines above this ("API returns 999999999999 when no limit is used").

Reply:
That's actually not the same thing, it is a client problem not directly related to the API and to the comment above, it was just supposed to be a really big number used as a default, big enough limit, so that you can iterate through all the items. I don't like this either, so I rewrote the logic so it's cleaner.